ui-admin/src/study/surveys/CreateSurveyModal.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useContext, useState } from 'react'
+import { StudyEnvContextT } from 'study/StudyEnvironmentRouter'
+import Modal from 'react-bootstrap/Modal'
+import LoadingSpinner from 'util/LoadingSpinner'
+import Api, { VersionedForm } from 'api/api'
+import { useNavigate } from 'react-router-dom'
+import { Store } from 'react-notifications-component'
+import { failureNotification } from '../../util/notifications'
+import { PortalContext, PortalContextT } from '../../portal/PortalProvider'
+
+// TODO: Add JSDoc
+// eslint-disable-next-line jsdoc/require-jsdoc
+const CreateSurveyModal = ({ studyEnvContext, isReadOnlyEnv, show, setShow }: {studyEnvContext: StudyEnvContextT,
+  isReadOnlyEnv: boolean, show: boolean, setShow:  React.Dispatch<React.SetStateAction<boolean>> }) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const [surveyName, setSurveyName] = useState('')
+  const [surveyStableId, setSurveyStableId] = useState('')
+
+  const portalContext = useContext(PortalContext) as PortalContextT
+  const navigate = useNavigate()
+
+  const createSurvey = async () => {
+    setIsLoading(true)
+    const createdSurvey = await Api.createNewSurvey(studyEnvContext.portal.shortcode,
+      {
+        createdAt: 0, id: '', lastUpdatedAt: 0, version: 1,
+        content: '{"pages":[]}', name: surveyName, stableId: surveyStableId
+      }).catch(e =>
+      Store.addNotification(failureNotification(`Error creating survey: ${e.message}`))
+    ) as VersionedForm
+
+    try {
+      await Api.createConfiguredSurvey(studyEnvContext.portal.shortcode,
+        studyEnvContext.study.shortcode,
+        studyEnvContext.currentEnv.environmentName,
+        {
+          allowAdminEdit: true,
+          allowParticipantReedit: true,
+          allowParticipantStart: true,
+          id: '',
+          prepopulate: false,
+          recurrenceIntervalDays: 0,
+          recur: false,
+          studyEnvironmentId: studyEnvContext.currentEnv.id,
+          survey: createdSurvey,
+          surveyId: createdSurvey.id,
+          surveyOrder: studyEnvContext.currentEnv.configuredSurveys.length
```

Comment:
As we continue to build out the survey builder, these options should eventually be exposed to the study admins for customization.